docs: Add new section "Running a Node"

amaurymartiny opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

ref: #7657 
ref: #7403 

[rendered](https://github.com/cosmos/cosmos-sdk/tree/am-7657-run-node/docs/run-node)

- Move `interfaces/keyring.md` to `run-node/keyring.md`, and update content.
- Move `using-the-sdk/quickstart.md` to `run-node/run-node.md` and `run-node/interact-node.md`, and update content.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

robert-zaremba commented on docs/run-node/interact-node.md:
````diff
@@ -0,0 +1,49 @@
+Now that your chain is running, it is time to try sending tokens from the first account you created to a second account. In a new terminal window, start by running the following query command:
+
+```bash
+simd query account $(simd keys show my_validator -a) --chain-id my-test-chain
````
would be good to use variables instead repeating the sub-commands:
```
export MY_VALIDATOR=$(....)

alessio replied:
Yeah I agree. I'd call it `MY_VALIDATOR_ADDRESS` for the sake of clarity. The gist of it is that shorter examples are better than longer ones.

robert-zaremba commented on docs/run-node/interact-node.md:
````diff
@@ -0,0 +1,49 @@
+You should see the current balance of the account you created, equal to the original balance of `stake` you granted it minus the amount you delegated via the `gentx`. Now, create a second account:
+
+```bash
+simd keys add receiver
````
```suggestion
simd keys add receiver
export RECEIVER=$(simd keys show receiver -a)
```

alessio replied:
We don't need to use the `export` statement though. For the sake of clarity and if we assume that the user runs all the commands in one terminal window/session, we can remove `export`.

robert-zaremba commented on docs/run-node/interact-node.md:
````diff
@@ -0,0 +1,49 @@
+simd keys add receiver
+```
+
+The command above creates a local key-pair that is not yet registered on the chain. An account is registered the first time it receives tokens from another account. Now, run the following command to send tokens to the second account:
````
```suggestion
The command above creates a local key-pair that is not yet registered on the chain. An account is created the first time it receives tokens from another account. Now, run the following command to send tokens to the `receiver` account:
```

alessio replied:
I'd call it the `recipient`. Not a big deal either way

alessio replied:
FTR `recipient` belongs to a finance-friendly terminology, so...

robert-zaremba commented on docs/run-node/interact-node.md:
````diff
@@ -0,0 +1,49 @@
+The command above creates a local key-pair that is not yet registered on the chain. An account is registered the first time it receives tokens from another account. Now, run the following command to send tokens to the second account:
+
+```bash
+simd tx send $(simd keys show my_validator -a) $(simd keys show receiver -a) 1000stake --chain-id my-test-chain
````
```suggestion
simd tx send $MY_VALIDATOR  $RECEIVER 1000stake --chain-id my-test-chain
```

robert-zaremba commented on docs/run-node/interact-node.md:
````diff
@@ -0,0 +1,49 @@
+
+```bash
+simd query account $(simd keys show receiver -a) --chain-id my-test-chain
+```
````
we could paste this command in the code block above, would make the whole think shorter.

robert-zaremba commented on docs/run-node/keyring.md:
```diff
@@ -78,15 +82,63 @@ $ pass init <GPG_KEY_ID>
+      --home string   directory for config and data (default "/Users/{your_username}/.simapp")
+      --trace         print out full stack trace on errors
+
+Use "simd keys [command] --help" for more information about a command.
```
do we want to copy the whole help output? This will mean that every time we change it we will need to remember to update this part. If we want to do it then I would prefer to have an appropriate note in the code to also update the docs.
Otherwise this should be enough:
Use `simd keys` for help about the `keys` command and `simd keys [command] --help` for more information.

alessio replied:
Plus, we should really instruct users on how to use our autocompletion command:

```
simd completion
```

At the start of a new **bash** session, users can just run:

```
. <(simd completion)
```

robert-zaremba commented on docs/run-node/run-node.md:
````diff
@@ -0,0 +1,90 @@
+simd add-genesis-account $(simd keys show my_validator -a) 100000000stake --chain-id my-test-chain
+```
+
+Note that the tokens in the SDK have the `{amount}{denom}` format.
````
Please explain denom. I was initially confused, because I was usually used in context of value denomination. Example: 1USD denominated in US Cent is 100 USCent. Or 1 BTC is denominated in satoshi, where 1 BTC = 1e9 satoshi.

clevinson replied:
We should also explain how "stake" just happens to be the denom in simapp, and for your own chain with its own staking denom, that should be used instead if following this guide.

robert-zaremba commented on docs/run-node/run-node.md:
```diff
@@ -0,0 +1,90 @@
+
+A `gentx` does three things:
+
+1. Makes the `validator` account you created into a validator operator account (i.e. the account that controls the validator).
```
```suggestion
1. Registers the `validator` account you created as a validator operator account (i.e. the account that controls the validator).
```

robert-zaremba commented on docs/run-node/run-node.md:
````diff
@@ -0,0 +1,90 @@
+simd gentx --help
+```
+
+Now that everything is set up, you can finally start your node:
````
How about moving this this paragraph into the section below (Run a localnet)?

robert-zaremba reviewed (commented):
Left few comments and suggestions.

clevinson commented on docs/run-node/keyring.md:
```diff
@@ -1,28 +1,32 @@
-# The keyring
+# Setting up the keyring
 
 This document describes how to configure and use the keyring and its various backends for an [**application**](../basics/app-anatomy.md). A separate document for implementing a CLI for an SDK [**module**](../building-modules/intro.md) can be found [here](#../building-modules/module-interfaces.md#cli). {synopsis}
```
```suggestion
This document describes how to configure and use the keyring and its various backends for an [**application**](../basics/app-anatomy.md). {synopsis}
```
Can we drop the last sentence? I'm not sure if its necessary.

clevinson commented on docs/run-node/keyring.md:
```diff
@@ -78,15 +82,63 @@ $ pass init <GPG_KEY_ID>
+
+This command generates a new 24-word mnemonic phrase, persists it to the relevant backend, and outputs information about the keypair. If this keypair will be used to hold value-bearing tokens, be sure to write down the mnemonic phrase somewhere safe!
+
+By default, the keyring generates a `secp256k1` keypair. The SDK also supports `ed25519` keys, which may be created by passing the `--algo ed25519` flag.
```
Just for my own clarification - is it required that a given app chain use one algo for all accounts? Or alternatively, is it possible to have some accounts on a chain have corresponding secp256k1 keys, and other accounts have corresponding ed25519 keys?

amaurymartiny replied:
Right now, `x/auth`'s antehandlers support these 2 algos on a same chain.

clevinson commented on docs/run-node/run-node.md:
```diff
@@ -0,0 +1,90 @@
+
+# Running a Node
+
+Now that the application is ready and the keyring populated, it's time to see how to run the blockchain node. In this section, the application we are running is called [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/v0.40.0-rc1/simapp), and its corresponding CLI binary `simd`. {synopsis}
```
```suggestion
Now that the application is ready and the keyring populated, it's time to see how to run the blockchain node. In this section, the application we are running is called [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/v0.40.0-rc2/simapp), and its corresponding CLI binary `simd`. {synopsis}
```

clevinson reviewed (approved):
Few minor comments, but otherwise lgtm!

amaurymartiny commented on docs/run-node/run-node.md:
```diff
@@ -0,0 +1,90 @@
+      |- config.toml                # Tendermint-related configuration file.
+      |- genesis.json               # The genesis file.
+      |- node_key.json              # Tendermint node pubkey that will be used for signing blocks.
+      |- priv_validator_key.json    # TODO
```
Does someone have a link to what this file does? In particular, I don't know the difference between this file and the `node_key.json` one.

alessio replied:
```
# Path to the JSON file containing the private key to use as a validator in the consensus protocol
priv_validator_key_file = "config/priv_validator_key.json"

# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
node_key_file = "config/node_key.json"
```

amaurymartiny commented:
Thanks a bunch @robert-zaremba @alessio @clevinson for this review, it was very helpful! I believe I addressed all comments, would appreciate another round of review.
